x/group/keeper/keeper.go

```diff
@@ -363,6 +363,7 @@ func (k Keeper) PruneProposals(ctx sdk.Context) error {
 		return nil
 	}
 	for _, proposal := range proposals {
+		proposal := proposal
```

Comment:
Resolves this lint warning:

```
x/group/keeper/keeper.go:375:18: G601: Implicit memory aliasing in for loop. (gosec)
				TallyResult: &proposal.FinalTallyResult,
```

Inspired by the implementation on Cosmos SDK main